libs/dh/admin/data-access-api/src/lib/dh-admin-user-role-audit-logs-data-access-api.store.ts

```diff
@@ -83,16 +86,49 @@ export class DhAdminUserRoleAuditLogsDataAccessApiStore extends ComponentStore<D
   );
 
   private assignAuditLogs = (response: MarketParticipantUserRoleAuditLogsDto) => {
-    const auditLogs: DhRoleAuditLogEntry[] = response.auditLogs.map((entry) => ({
-      entry: {
-        ...entry,
-        changedValueTo: mapChangeDescriptionJson(
-          entry.userRoleChangeType,
-          this.parseChangeDescriptionJson(entry.changeDescriptionJson)
-        ),
-      },
-      timestamp: entry.timestamp,
-    }));
+    const auditLogs: DhRoleAuditLogEntry[] = response.auditLogs
+      .filter((x) => x.userRoleChangeType !== 'PermissionsChange')
+      .map((entry) => ({
+        entry: {
+          ...entry,
+          changedValueTo: mapChangeDescriptionJson(
+            entry.userRoleChangeType,
+            this.parseChangeDescriptionJson(entry.changeDescriptionJson)
+          ),
+        },
+        timestamp: entry.timestamp,
+      }));
+
+    const orderedPermissionChanges = response.auditLogs
+      .filter((entry) => entry.userRoleChangeType === 'PermissionsChange')
+      .sort((a, b) => new Date(a.timestamp).getTime() - new Date(b.timestamp).getTime());
+
+    orderedPermissionChanges.forEach((entry, i) => {
+      const prevPermissions =
+        i > 0
+          ? (JSON.parse(orderedPermissionChanges[i - 1].changeDescriptionJson)
+              .Permissions as number[])
+          : [];
+      const currPermissions = JSON.parse(entry.changeDescriptionJson).Permissions as number[];
+
+      const permissionChagnes: { type: ChangeType; perms: number[] }[] = [
```

Comment:
nitpick: Spelling error in "permissionChagnes"

Reply:
Fixed